instill/model/v1alpha/model.proto

```diff
@@ -0,0 +1,372 @@
+  float score = 2;
+}
+
+// ClassificationOutputs represents a list of classification task outputs
+message ClassificationOutputs {
+  // A list of classification outputs
+  repeated ClassificationOutput classification_outputs = 1;
+}
+
+// BoundingBox represents the bounding box data structure
+message BoundingBox {
+  // Bounding box top y-axis value
+  float top = 1;
+  // Bounding box left x-axis value
+  float left = 2;
+  // Bounding box width value
+  float width = 3;
+  // Bounding box height value
+  float height = 4;
+}
+
+// BoundingBoxObject represents a predicted bounding box object
+message BoundingBoxObject {
+  // Bounding box object category
+  string category = 1;
+  // Bounding box object scroe
+  float score = 2;
+  // Bounding box
+  BoundingBox bounding_box = 3;
+}
+
+// DetectionOutput represents the output of detection task
+message DetectionOutput {
+  // A list of buonding box objects
+  repeated BoundingBoxObject bounding_box_objects = 1;
+}
+
+// DetectionOutputs represents a list of detection task outputs
+message DetectionOutputs {
+  // A list of detection outputs
+  repeated DetectionOutput detection_output = 1;
+}
+
+// Model service responds to incoming model requests.
+service ModelService {
+
+  // Liveness method receives a LivenessRequest message and returns a
+  // LivenessResponse message.
+  // See https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+  rpc Liveness(LivenessRequest) returns (LivenessResponse) {
+    option (google.api.http) = {
+      get : "/__liveness"
+      additional_bindings : [ {get : "/health/model"} ]
+    };
+  }
+
+  // Readiness method receives a ReadinessRequest message and returns a
+  // ReadinessResponse message.
+  // See https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+  rpc Readiness(ReadinessRequest) returns (ReadinessResponse) {
+    option (google.api.http) = {
+      get : "/__readiness"
+    };
+  }
+
+  // CreateModel method receives a CreateModelRequest message and returns
+  // a CreateModelResponse message.
+  rpc CreateModel(CreateModelRequest) returns (CreateModelResponse) {
+    option (google.api.http) = {
+      post : "/models"
+      body : "*"
+    };
+  }
+
+  // CreateModel method receives a CreateModelRequest message and returns
+  // a CreateModelResponse message.
+  rpc ListModel(ListModelRequest) returns (ListModelResponse) {
+    option (google.api.http) = {
+      get : "/models"
+    };
+  }
+
+  // GetModel method receives a GetModelRequest message and returns
+  // a GetModelResponse message.
+  rpc GetModel(GetModelRequest) returns (GetModelResponse) {
+    option (google.api.http) = {
+      get : "/models/{name}"
+    };
+  }
+
+  // UpdateModel method receives a UpdateModelRequest message and returns
+  // a UpdateModelResponse message.
+  rpc UpdateModel(UpdateModelRequest) returns (UpdateModelResponse) {
+    option (google.api.http) = {
+      patch : "/models/{model.name}/versions/{version}"
```

Comment:
@pinglin this RPC seem to be an update model version RPC, what do you think?

Reply:
Right. That's more specific.